src/components/Main/MainView/MessageInput/use/textFieldMirror.ts

```diff
@@ -0,0 +1,120 @@
+const properties = [
+  'position:absolute;',
+  'overflow:auto;',
+  'word-wrap:break-word;',
+  'top:0px;',
+  'left:-9999px;'
+]
+
+// カーソルの位置に影響するCSSプロパティをテキストフィールドからdiv要素にコピーする
+const propertyNamesToCopy = [
+  'box-sizing',
+  'font-family',
+  'font-size',
+  'font-style',
+  'font-variant',
+  'font-weight',
+  'height',
+  'letter-spacing',
+  'line-height',
+  'max-height',
+  'min-height',
+  'padding-bottom',
+  'padding-left',
+  'padding-right',
+  'padding-top',
+  'border-bottom',
+  'border-left',
+  'border-right',
+  'border-top',
+  'text-decoration',
+  'text-indent',
+  'text-transform',
+  'width',
+  'word-spacing'
+]
+
+// { テキストフィールド: ミラー }
+const mirrorMap = new WeakMap()
+
+// テキストフィールドを模したdiv要素を画面外に準備する
+//
+// textField: HTMLInputElement or HTMLTextAreaElement element
+// markerPosition: 再現したいカーソルのインデックス(デフォルト値: テキスト終端)
+//
+// mirror: DOMに紐付いたエレメントを返す. caret位置取得後のエレメント削除は呼び出し元の責任
+// marker: カーソルと同じ位置に存在するspanエレメント
```

Comment:
```suggestion
/**
 * テキストフィールドを模したdiv要素を画面外に準備する
 * @param markerPosition 再現したいカーソルのインデックス (nullのときはテキスト終端)
 */
```
返り値の型抽出してそこにmirrorとmarkerの説明をTSDocで書いてほしい 🙏